build_scripts/api/dreambooth/utils/model_utils.py

```
@@ -0,0 +1,278 @@
+        from transformers import CLIPTextModel
+
+        return CLIPTextModel
+    elif model_class == "RobertaSeriesModelWithTransformation":
+        from diffusers.pipelines.alt_diffusion.modeling_roberta_series import RobertaSeriesModelWithTransformation
+
+        return RobertaSeriesModelWithTransformation
+    else:
+        raise ValueError(f"{model_class} is not supported.")
+
+
+# from modules.sd_models import CheckpointInfo
+
+def get_checkpoint_match(search_string):
+    try:
+        from modules import sd_models
+        for info in sd_models.checkpoints_list.values():
+            if search_string in info.title or search_string in info.model_name or search_string in info.filename:
+                return info
+    except:
+        pass
+    return None
+
+
+def disable_safe_unpickle():
+    try:
+        from modules import shared as auto_shared
+        auto_shared.cmd_opts.disable_safe_unpickle = True
+        torch.load = unsafe_torch_load
+    except:
+        pass
+
+
+def enable_safe_unpickle():
+    try:
+        from modules import shared as auto_shared
+        auto_shared.cmd_opts.disable_safe_unpickle = False
+        torch.load = load
+    except:
+        pass
+
+
+def xformerify(obj, try_sdp=True):
+    if try_sdp:
+        try:
+            from diffusers.models.attention_processor import AttnProcessor2_0
+            print("Enabling SDP")
+            obj.set_attn_processor(AttnProcessor2_0())
+            return
+        except:
+            pass
+
+    if is_xformers_available():
+        try:
+            print("Enabling xformers memory efficient attention for unet")
+            obj.enable_xformers_memory_efficient_attention()
+        except ModuleNotFoundError:
+            print("xformers not found, using default attention")
+
+
+def torch2ify(unet):
+    if hasattr(torch, 'compile'):
+        try:
+            unet = torch.compile(unet, mode="max-autotune", fullgraph=False)
+            print("Compiled unet")
+        except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```